crates/bevy_diagnostic/src/diagnostic.rs

```diff
@@ -45,9 +45,7 @@ impl Diagnostic {
     /// Add a new value as a [`DiagnosticMeasurement`]. Its timestamp will be [`Instant::now`].
     pub fn add_measurement(&mut self, value: f64) {
         let time = Instant::now();
-        if self.max_history_length > 1 {
-            if self.history.len() == self.max_history_length {
-                if let Some(removed_diagnostic) = self.history.pop_front() {
+        if self.max_history_length > 1 && self.history.len() == self.max_history_length && let Some(removed_diagnostic) = self.history.pop_front() {
```

Comment:
`if let chain` is an unstable feature : https://github.com/rust-lang/rust/issues/53667

You must use separate the `if let` from the other conditions.

Reply:
Very well

Reply:
Pardon me,my bad